common/auth/federation_client.go

```diff
@@ -108,10 +112,12 @@ func (c *x509FederationClient) renewSecurityToken() (err error) {
 		return fmt.Errorf("failed to refresh leaf certificate: %s", err.Error())
 	}
 
-	updatedTenancyID := extractTenancyIDFromCertificate(c.leafCertificateRetriever.Certificate())
-	if c.tenancyID != updatedTenancyID {
-		err = fmt.Errorf("unexpected update of tenancy OCID in the leaf certificate. Previous tenancy: %s, Updated: %s", c.tenancyID, updatedTenancyID)
-		return
+	if c.refresh {
```

Comment:
Looks like this boolean variable doesn't switch if some objects are refreshed.  Instead, it just controls if we validate the tenancy ID.

Now, given certificate retrievers control if certificates should be refreshed or not.  Probably we want to use one boolean variable to enable/disable refreshing certificates and tenancy ID validation.  In this case, StaticX509CertificateRetriever wouldn't be necessary.  (Honestly, though I like the static retriever, rather than boolean variable.)